File: apps/HSMS-fe/src/app/models/state.dto.ts
// src/models/state.dto.ts
export interface StateMastersDto {
   id: number;
  countryId: number;
  countryName: string;
  name: string;
  stateCode: string;
  createdAt: string;
  updatedAt: string;
}


export interface CreateStateMasters {  
           // string? stateID  
  stateName?: string;            // string? StateName
  stateCode?: string;            // int? StateCode
  countryID?: number;            // int? CountryID
  
}

export interface UpdateStateMasters {
   id?: number;               // string? stateID  
  stateName?: string;            // string? StateName
  stateCode?: string;            // int? StateCode
  countryID?: number;            // int? CountryID
  
}

File: libs/contracts/Dto/StateDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HSMS.contracts.Dto
{
    public class StateDto
    {
        public int Id { get; set; }
        public int CountryId { get; set; }
        public string CountryName { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string? StateCode { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset? UpdatedAt { get; set; }
    }
}
